src/base.jl

```diff
@@ -280,15 +284,116 @@ function add_component!(sys::ConcreteSystem, component::T) where T <: Component
     return nothing
 end
 
+function _convert_forecasts!(system_forecasts::SystemForecasts,
+                             old_forecasts::SystemForecastsDeprecated)
+    for (sym, forecasts) in old_forecasts
+        for forecast in forecasts
+            add_forecast!(system_forecasts, forecast)
+        end
+    end
+
+    if !_validate_component_label_uniqueness(system_forecasts)
+        throw(DataFormatError("components and labels are not unique within forecast array"))
+    end
+    return system_forecasts
+end
+
+function _validate_component_label_uniqueness(system_forecasts::SystemForecasts)::Bool
+    match = true
+
+    for (issue_time, forecasts) in system_forecasts
+        unique_components = Set{Tuple{<:Component, String}}()
+        num_not_unique = 0
+        for forecast in forecasts
+            component_label = (forecast.component, forecast.label)
+            if component_label in unique_components
+                num_not_unique += 1
+                match = false
+                @error("not all components in forecast vector are unique", component_label,
+                       issue_time)
+            else
+                push!(unique_components, component_label)
+            end
+        end
+        @info "summary" num_not_unique length(forecasts)
+    end
+
+    return match
+end
+
+function add_forecast!(forecasts::SystemForecasts, issue_time::IssueTime,
+                       forecast::Forecast)
+    if !haskey(forecasts, issue_time)
+        forecasts[issue_time] = Forecasts{Forecast}()
+    end
+
+    push!(forecasts[issue_time], forecast)
+end
+
+function add_forecast!(forecasts::SystemForecasts, forecast::Forecast)
+    issue_time = get_issue_time(forecast)
+    add_forecast!(forecasts, issue_time, forecast)
+end
+
 """
 Args:
     A ConcreteSystem struct
     A :Symbol=>Array{ <: Forecast,1} Pair denoting the forecast name and array of device forecasts
 Returns:
     A System struct with a modified forecasts field
 """
-function add_forecast!(sys::ConcreteSystem,fc::Pair{Symbol,Array{Forecast,1}})
-    sys.forecasts[fc.first] = fc.second
+function add_forecast!(sys::ConcreteSystem, fc::Pair{Symbol, Vector{Forecast}})
```

Comment:
We do need a new add_forecast function.  Should it accept System and a vector of forecasts?  We can determine IssueTime and then store accordingly.  Do we need a function that accepts one forecast?

Reply:
Given that we have an `add_component` function, I think we should have an `add_forecast!` function that accepts a single forecast. Otherwise, your proposal for arguments seems fine to me.